sonar/modules/deposits/api.py

```diff
@@ -126,42 +96,136 @@ def log_action(self, user, action, comment=None):
 
     def create_document(self):
         """Create document from deposit."""
-        metadata = {
-            'title': [{
-                'type':
-                'bf:Title',
-                'mainTitle': [{
-                    'language': self['metadata']['languages'][0],
-                    'value': self['metadata']['title']
-                }]
-            }],
-            'documentType':
-            self['metadata']['document_type'],
-            'language': [{
-                'type': 'bf:Language',
-                'value': language
-            } for language in self['metadata']['languages']],
-        }
-
-        for contributor in self['contributors']:
-            author = {'type': 'person', 'name': contributor['name']}
-            if contributor.get('affiliation'):
-                author['qualifier'] = contributor['affiliation']
-
-            metadata.get('authors', []).append(author)
-
-        if self['metadata'].get('etc'):
-            self['extent'] = self['metadata']['etc']
+        metadata = {}
+
+        # Document type
+        metadata['documentType'] = self['metadata']['documentType']
+
+        # Title
+        metadata['title'] = [{
+            'type':
+            'bf:Title',
+            'mainTitle': [{
+                'language': self['metadata']['language'],
+                'value': self['metadata']['title']
+            }]
+        }]
+
+        # Subtitle
+        if self['metadata'].get('subtitle'):
+            metadata['title'][0]['subtitle'] = [{
+                'language':
+                self['metadata']['language'],
+                'value':
+                self['metadata']['subtitle']
+            }]
 
-        for key, abstract in enumerate(self['metadata']['abstracts']):
-            metadata.get('abstracts', []).append({
+        # Other title
+        if self['metadata'].get('otherLanguageTitle', {}).get('title'):
+            metadata['title'][0]['mainTitle'].append({
                 'language':
-                self['metadata']['languages'][key]
-                if self['metadata']['languages'][key] else 'eng',
+                self['metadata']['otherLanguageTitle']['language'],
```

Comment:
are you sure to always have a 'language' ? Maybe better to use `self['metadata']['otherLanguageTitle'].get('language')` to avoid a KeyError ?